### ISSUE TITLE
Maliit Causes Plasma to Crash

### ISSUE THREAD

**Issue by FedX-sudo** [maliit/keyboard#59]
## System:

- MS Surface Pro 2
  -  Intel Core i5-4200U
  -  Memory 4GB 
  -  Storage 128 GB
- OS:
  - Base system of Pop!_os 21.04
  - Hijacked with Bedrock Linux
  - Running Alpine Linux modified to PostmarketOS
  - DE: Plasma Mobile
  - Kernel: 5.11.0-7620-generic (from Pop!_os)

## Issue:

Maliit does not always show up which has been a previously reported issue. I therefore plug in a USB keyboard, and Plasma deactivates Maliit, unplugging the keyboard causes Plasma to deactivate Maliit. Sometimes hot-plugging the keyboard will crash Plasma, and sometimes Maliit will not show up, at all. This could be a Plasma issue, but it is directly tied to the virtual keyboard.

**Comment by hedmo** [comment]
i am facing the same issue.  for me it started at plasma 5.22.x. i am running maliit in gentoo plasma desktop on a asus transformer that has a removable keyboard . even if i dont detach the keyboard and just activate the virtual keyboard plasma can crash.

**Comment by FedX-sudo** [comment]
Glad to know its not just me. Out of curiosity (and hopefully this can help), what is the system? Specifically the base system and the kernel.

**Comment by hedmo** [comment]
Base system is Gentoo  and kernel version is 5.14.7->5.15.0-rc3

**Comment by FedX-sudo** [comment]
Hmm, not sure. Are you on X11 or Wayland?

**Comment by hedmo** [comment]
Maliit-keyboard only works on wayland if i understand it correct.

**Comment by FedX-sudo** [comment]
Alright. Let me try to figure out how to pull some logs. Not sure exactly how I should do that...

**Comment by dobey** [comment]
> Maliit-keyboard only works on wayland if i understand it correct.

It should work on both. Going to need a stack trace, as it's working fine on Plasma Mobile.

**Comment by FedX-sudo** [comment]
I am seeing the issue on Plasma Mobile, I simply have running on an x86_64 tablet, perhaps its an x86 issue or an issue with the large landscape layout?

**Comment by hedmo** [comment]
The landscape is an issues for me to. Sometimes it does not scale correct from picture mode to landscape mode. But if this can affect maliit-keyboard i DO not know.

**Comment by dobey** [comment]
> I am seeing the issue on Plasma Mobile, I simply have running on an x86_64 tablet, perhaps its an x86 issue or an issue with the large landscape layout?

I haven't noticed any crashing of the shell when using the keyboard on my Pinephone with the Manjaro Plasma Mobile dev image. What versions of maliit-framework and maliit-keyboard are you using?

**Comment by FedX-sudo** [comment]
I am using 2.0 from PostMarketOS.

**Comment by dobey** [comment]
> I am using 2.0 from PostMarketOS.

And what version of Plasma? Can you verify if the issues still occur on latest releases and/or git versions?

**Comment by FedX-sudo** [comment]
PostMarketOS UI Plasma Mobile 3.3

**Comment by FedX-sudo** [comment]
I am on the edge branch, which is unstable rolling, so I would assume its 5.22.

**Comment by FedX-sudo** [comment]
I just upgraded my system, and am now on Plasma 5.23. It appears the issue was fixed in this version by simply not killing Maliit when hot plugging a keyboard. Before this, I was doing some debugging, and I believe the underlying issue is with lines 338-334 of [keyboard/qml/keys/CharKey.qml](https://github.com/maliit/keyboard/blob/master/qml/keys/CharKey.qml):

```
Connections {
    target: swipeArea.drag
    onActiveChanged: {
        if (swipeArea.drag.active)
            keyMouseArea.cancelPress();
    }
}
```

When running `maliit-framework` in the terminal I get an error saying that syntax is depreciated, and I believe Plasma does not properly handle that error and crashed. Thanks for your work and help!
